### DISCUSSION TITLE
The following relation ids were removed from the list because they are missing or invalid: {the actual value of the course field}

### DISCUSSION THREAD

**Discussion by Ilyas-Codes** [pocketbase/pocketbase#4172] [Q&A] [answered]
Hey everyone, thank you Gani for the work you're doing on Pocketbase. so I have an `onRecordAfterUpdateRequest` hook that creates or deletes records on my `progress` collection whenever the `assignees` relation field (this field is multiple and points to my `users` collection) in my `modules` collection is updated, so whenever a user/assignee is added or removed from the `assignees` field, I either create or delete records on my `progress` collection. the `progress` collection has 3 fields, `course` a relation field (single) that points to the `modules` collection, `assignee` a relation field (single) that points to the `users` collection and `status` a select field with 3 values. everything on my hook works fine except the `course` field in the `progress` collection which I'm setting its value to the `title` field of the record being updated in the `modules` collection like so: 
```
const updatedRecord = e.record;
course: updatedRecord.getString("title"),
```
when a `progress` record is created, the course field shows the title as expected but when I open up the record, the course field is cleared and there is an exclamation mark to the right of the field that says the warning I put in the title.

here is the hook:
```
onRecordAfterUpdateRequest((e) => {
  const updatedRecord = e.record;
  const originalRecord = updatedRecord.originalCopy();
  const originalAssignees = originalRecord.getStringSlice("assignees");
  const updatedAssignees = updatedRecord.getStringSlice("assignees");

  const newAssignees = updatedAssignees.filter(
    (assignee) => !originalAssignees.includes(assignee)
  );
  const removedAssignees = originalAssignees.filter(
    (assignee) => !updatedAssignees.includes(assignee)
  );

  const progressCollection = $app.dao().findCollectionByNameOrId("progress");

  newAssignees.forEach((assignee) => {
    const progressRecord = new Record(progressCollection, {
      course: updatedRecord.getString("title"),
      assignee: assignee,
      status: "Not Started",
    });

    $app.dao().saveRecord(progressRecord);
  });

  removedAssignees.forEach((assignee) => {
    const progressRecord = $app
      .dao()
      .findFirstRecordByData(progressCollection.name, "assignee", assignee);

    if (progressRecord) {
      $app.dao().deleteRecord(progressRecord);
    }
  });
}, "modules");
```

**Comment by ganigeorgiev** [comment]
I'm not sure that I fully understand your collections structure but in the above for `course` you are assigning the `title` field where based on your description this field is a `relation`, meaning that it needs a valid `course` relation id (in your case probably `updatedRecord.id`).

**Comment by Ilyas-Codes** [comment]
@ganigeorgiev no, the `title` field is a string (course title example)

**Reply by ganigeorgiev** [reply]
That's OK, but the `progress.course` field is a `relation` (at least based on your description).

**Reply by Ilyas-Codes** [reply]
exactly yes, this the structures of both collections:

`modules` collection has 4 fields only 2 are relevant here: `title` a text/string and `assignees` a relation field (multiple) that points to the `users` collection (the default auth collection that PB comes with)

progress collection has 3 fields: `course` a relation field (single) that points to the `modules` collection , `assignee`  a relation field (single) that points to the `users` collection and `status` which is just a select field

**Reply by ganigeorgiev** [reply, marked as answer]
It seems that we are going in circles but I'll try one last time with a direct example. Try to replace:

```js
course: updatedRecord.getString("title"),
```

with:

```js
course: updatedRecord.id,
```

(_`relation` field always expect id(s)_).

**Reply by Ilyas-Codes** [reply]
yup, that worked like a charm, I'm just dumb lol. have a good day!

**Reply by ganigeorgiev** [reply]
_In a future version app level db constraint will be always checked on `saveRecord()` (currently to perform the record validations you need to use the [form model](https://pocketbase.io/docs/js-records/#create-new-record-with-data-validations))._
